crates/foundry/evm/core/src/decode.rs

```diff
@@ -229,10 +227,12 @@ fn trimmed_hex(s: &[u8]) -> String {
     if s.len() <= n {
         hex::encode(s)
     } else {
+        let start = s.get(..n / 2).unwrap_or(s);
+        let end = s.get(s.len().saturating_sub(n / 2)..).unwrap_or(s);
```

Comment:
i'd rather use expect here. it's a bit weird to fall back to the string when checked the length before